src/ucm/mmap/install.c

```diff
@@ -16,28 +16,64 @@
 #include <ucm/util/log.h>
 #include <ucm/util/reloc.h>
 #include <ucm/util/sys.h>
+#include <ucm/bistro/bistro.h>
 #include <ucs/sys/math.h>
+#include <ucs/sys/checker.h>
 
 #include <sys/mman.h>
 #include <sys/shm.h>
 #include <unistd.h>
 #include <pthread.h>
 
+#define UCM_IS_HOOK_ENABLED(_entry)                                \
+     (((_entry)->hook_type == UCM_HOOK_FORCE_RELOC) ||             \
+      ((ucm_global_opts.mmap_hook_mode == UCM_MMAP_HOOK_RELOC) &&  \
+       ((_entry)->hook_type & UCM_HOOK_RELOC)) ||                  \
+      ((ucm_global_opts.mmap_hook_mode == UCM_MMAP_HOOK_BISTRO) && \
+       ((_entry)->hook_type & UCM_HOOK_BISTRO)))
+
+
+#define UCM_HOOK_STR(_entry)                                      \
+    (((ucm_global_opts.mmap_hook_mode == UCM_MMAP_HOOK_RELOC) ||  \
+      ((_entry)->hook_type == UCM_HOOK_FORCE_RELOC)) ?            \
+      "relocation table entry" : "bistro hook")
+
+#if HAVE_DECL_SYS_SHMAT
+#  define UCM_SHMAT_MODE UCM_HOOK_BOTH
+#else
+#  define UCM_SHMAT_MODE UCM_HOOK_FORCE_RELOC
+#endif
+
+#if HAVE_DECL_SYS_SHMDT
+#  define UCM_SHMDT_MODE UCM_HOOK_BOTH
+#else
+#  define UCM_SHMDT_MODE UCM_HOOK_FORCE_RELOC
+#endif
+
+typedef enum ucm_mmap_hook_type {
+    UCM_HOOK_FORCE_RELOC = 0, /* special case when BISTRO could not be used due to
```

Comment:
really, shmdt/shmat on powerpc is missing?? sounds wierd.. what is glibc on ppc64 doing for shmat/dt then?

Also, what would be the difference between FORCE_RELOC and specifying a single flag RELOC?
the hook mode of every call could be the OR of all allowed modes, and if only one flag is specified - means it's "force" because it's the only allowed mode.